test/integration/src/main/java/io/pravega/test/integration/selftest/Actor.java

```diff
@@ -81,10 +82,18 @@ protected void doStart() {
         notifyStarted();
         this.runTask = Futures
                 .delayedFuture(INITIAL_DELAY, this.executorService)
-                .thenCompose(v -> run());
+                .thenCompose(v -> runWithException());
         this.runTask.whenComplete((r, ex) -> stopAsync());
     }
 
+    private CompletableFuture<Void> runWithException() {
+        try {
+            return run();
+        } catch (TxnFailedException ex) {
+            throw new RuntimeException(ex);
```

Comment:
Methods that return a CP should not throw but return a failed CompletableFuture

Reply:
Yes Enrico